src/test/functional/functional-tests.js

```diff
@@ -2512,4 +2512,210 @@ describe('functional tests', function() {
       })
 
     })})
+
+
+  describe('Object Name special characters test without Prefix', ()=> {
+    // Isolate the bucket/object for easy debugging and tracking.
+    const bucketNameForSpCharObjects = "minio-js-test-obj-spwpre-" + uuid.v4()
+    before((done) => client.makeBucket(bucketNameForSpCharObjects, '', done))
+    after((done) => client.removeBucket(bucketNameForSpCharObjects, done))
+
+    // Reference:: https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html
+    const objectNameSpecialChars="äöüex ®©µÄÆÐÕæŒƕƩǅ 01000000 0x40 \u0040 amȡȹɆple&0a!-_.*'()&$@=;:+,?<>.pdf"
+
+    const objectContents = Buffer.alloc(100 * 1024, 0)
+
+    describe('Without Prefix Test', function () {
+
+      step(`putObject(bucketName, objectName, stream)_bucketName:${bucketNameForSpCharObjects}, _objectName:${objectNameSpecialChars}, stream:100Kib_`, done => {
+        client.putObject(bucketNameForSpCharObjects, objectNameSpecialChars, objectContents)
+          .then(() => {
+            done()
+          })
+          .catch(done)
+      })
+
+      step(`listObjects(bucketName, prefix, recursive)_bucketName:${bucketNameForSpCharObjects}, prefix:"/, true,`, done => {
```

Comment:
```suggestion
      step(`listObjects(bucketName, prefix, recursive)_bucketName:${bucketNameForSpCharObjects}, prefix:"", true,`, done => {
```

Reply:
Thank you @harshavardhana . I have cleaned it up.